src/navigation/RootStackNavigation.tsx

```diff
@@ -231,7 +235,7 @@ const AppUnlockHandler = ({ children }: { children: ReactNode }) => {
       if (!wallet) {
         if (lastNavigationState) {
           // When we are at the wallet creation flow we want to reset to the last screen
-          setNavigationState(lastNavigationState)
+          restoreNavigation(navigation, lastNavigationState)
```

Comment:
Have a look at my comments on the implementation of this function later in this PR, I explain why I add the `navigation` object as prop.